File: apps/api/src/runsight_api/transport/schemas/steps.py
from typing import List, Optional

from pydantic import BaseModel, ConfigDict


class StepResponse(BaseModel):
    id: str
    name: str
    type: str
    path: str
    description: Optional[str] = None


class StepCreate(BaseModel):
    model_config = ConfigDict(extra="forbid")

    id: Optional[str] = None
    name: str
    type: str = "step"
    description: Optional[str] = None


class StepUpdate(BaseModel):
    model_config = ConfigDict(extra="forbid")

    name: Optional[str] = None
    type: Optional[str] = None
    description: Optional[str] = None


class StepListResponse(BaseModel):
    items: List[StepResponse]
    total: int

File: apps/gui/src/types/schemas/steps.ts
import { z } from "zod";

export const StepResponseSchema = z.object({
  id: z.string(),
  name: z.string(),
  type: z.string(),
  path: z.string(),
  description: z.string().nullable().optional(),
});
export type StepResponse = z.infer<typeof StepResponseSchema>;

export const StepCreateSchema = z.object({
  id: z.string().nullable().optional(),
  name: z.string(),
  type: z.string().default("step"),
  description: z.string().nullable().optional(),
});
export type StepCreate = z.infer<typeof StepCreateSchema>;

export const StepUpdateSchema = z.object({
  name: z.string().nullable().optional(),
  type: z.string().nullable().optional(),
  description: z.string().nullable().optional(),
});
export type StepUpdate = z.infer<typeof StepUpdateSchema>;

export const StepListResponseSchema = z.object({
  items: z.array(StepResponseSchema),
  total: z.number(),
});
export type StepListResponse = z.infer<typeof StepListResponseSchema>;
